### PR TITLE
Added circular ensembles

### PR DESCRIPTION
**Pull request by czgdp1807**
#### Brief description of what is fixed or changed
Circular ensembles have been added.

#### Other comments
More features will be added, if possible.
@Upabjojr @sidhantnagpal I am facing a problem(well the last option will be to keep it unevaluated) in exactly defining the density, since it is based on Haar measure which is not unique. Please let me know of any thing available in the literature so that I can do some improvements. I am also right now doing some searching on Google.

#### Release Notes

* stats
  * Circular ensembles have been added in `sympy.stats`.

### CODE REVIEW

**Comment by czgdp1807** [comment]
> @Upabjojr @sidhantnagpal I am facing a problem(well the last option will be to keep it unevaluated) in exactly defining the density, since it is based on Haar measure which is not unique. Please let me know of any thing available in the literature so that I can do some improvements. I am also right now doing some searching on Google.

@jksuom Any comments on this. I don't know much about Haar measure. What I know is that it is not unique, so how can I define the density of various circular ensembles. Thanks.

File: sympy/stats/random_matrix_models.py [hunk 1]
```diff
@@ -180,13 +183,108 @@ def density(self, expr):
...
+    Abstract class for Circular ensembles.
+    Contains the properties and methods
+    common to all the circular ensembles.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Circular_ensemble
+    """
+    def _compute_joint_eigen_dsitribution(self, beta):
```
Comment (jksuom) [inline sympy/stats/random_matrix_models.py]: dsitribution -> distribution

Comment (czgdp1807) [reply sympy/stats/random_matrix_models.py]: There are many other spelling mistakes. Thanks for pointing out. I will correct them in the next commit.

**Comment by czgdp1807** [comment]
@Upabjojr The work is complete on circular ensembles.
Regarding `Haar` measure, I will open a separate PR in continuation of this one as predicting things on it's implementation will be difficult for now.

**Pull request merged**